File: split_chapters.js
const fs = require('fs');
const path = require('path');

// 读取txt文件
const content = fs.readFileSync('4-正文/第1卷  初来乍到.txt', 'utf-8');

// 找到所有章节标题
const pattern = /^第(\d+)章.*$/gm;
const matches = [];
let match;

while ((match = pattern.exec(content)) !== null) {
    matches.push({
        num: parseInt(match[1]),
        title: match[0],
        start: match.index
    });
}

console.log(`找到 ${matches.length} 个章节`);

// 创建第13章及以后的文件
for (let i = 0; i < matches.length; i++) {
    const chapter = matches[i];
    const chapterNum = chapter.num;
    
    // 跳过前12章
    if (chapterNum <= 12) {
        console.log(`跳过第${chapterNum}章（已存在）`);
        continue;
    }
    
    // 计算结束位置
    let endPos;
    if (i < matches.length - 1) {
        endPos = matches[i + 1].start;
    } else {
        endPos = content.length;
    }
    
    // 提取章节内容
    const chapterContent = content.slice(chapter.start, endPos).trim();
    
    // 创建文件名
    const filename = `4-正文/第${chapterNum}章.md`;
    
    // 写入文件
    const lines = chapterContent.split('\n');
    let mdContent = `# ${chapter.title}\n\n`;
    
    // 跳过第一行（标题）
    if (lines.length > 1) {
        mdContent += lines.slice(1).join('\n').trim();
    }
    
    fs.writeFileSync(filename, mdContent, 'utf-8');
    console.log(`已创建: ${filename} - ${chapter.title}`);
}

console.log(`\n完成！共创建了 ${matches.length - 12} 个新章节文件`);

File: split_chapters.py
import re
import os

# 读取txt文件
with open(r'4-正文\第1卷  初来乍到.txt', 'r', encoding='utf-8') as f:
    content = f.read()

# 找到所有章节标题
pattern = r'^(第\d+章[^\n]*)'
chapters = []
matches = list(re.finditer(pattern, content, re.MULTILINE))

print(f"找到 {len(matches)} 个章节")

for i, match in enumerate(matches):
    chapter_title = match.group(1).strip()
    start_pos = match.start()
    
    # 计算章节内容的结束位置
    if i < len(matches) - 1:
        end_pos = matches[i + 1].start()
    else:
        end_pos = len(content)
    
    # 提取章节内容
    chapter_content = content[start_pos:end_pos].strip()
    
    # 章节编号
    chapter_num = i + 1
    
    # 保存为md文件
    if chapter_num <= 12:
        print(f"跳过第{chapter_num}章（已存在）")
        continue
    
    filename = f'4-正文/第{chapter_num}章.md'
    
    # 写入文件
    with open(filename, 'w', encoding='utf-8') as f:
        f.write(f"# {chapter_title}\n\n")
        # 添加章节内容，去掉标题行（因为已经在#中了）
        lines = chapter_content.split('\n')
        if len(lines) > 0:
            # 跳过第一行（章节标题在文件开头）
            f.write('\n'.join(lines[1:]).strip())
    
    print(f"已创建: {filename} - {chapter_title}")

print(f"\n完成！共创建了 {len(matches) - 12} 个新章节文件")